tests/scripts/github_mergebot.py

```diff
@@ -0,0 +1,520 @@
+            "",
+        )
+        return body
+
+    def body_with_co_authors(self) -> str:
+        """
+        Add 'Co-authored-by' strings to the PR body based on the prior commits
+        in the PR
+        """
+        body = self.processed_body()
+        author_lines = self.co_authors()
+        logging.info(f"Found co-authors: author_lines={author_lines}")
+        for author_line in author_lines:
+            if author_line not in body:
+                # If the line isn't already in the PR body (it could have been
+                # added manually), put it in
+                body = f"{body}\n\nCo-authored-by: {author_line}"
+
+        return body
+
+    def merge(self) -> None:
+        """
+        Request a merge of this PR via the GitHub API
+        """
+        url = f"pulls/{self.number}/merge"
+
+        title = self.raw["title"]
+        body = self.body_with_co_authors()
+        self.lint_commit_message(title, body)
+        logging.info(f"Full commit:\n{title}\n\n{body}")
+
+        data = {
+            "commit_title": title,
+            "commit_message": body,
+            # The SHA is necessary in case there was an update right when this
+            # script ran, GitHub will sort out who won
+            "sha": self.head_oid(),
+            "merge_method": "squash",
+        }
+        if self.dry_run:
+            logging.info(f"Dry run, would have merged with url={url} and data={js(data)}")
+            return
+
+        self.github.put(url, data=data)
+
+    def merge_requested(self) -> bool:
+        """
+        Check if this PR has had a merge requested
+        """
+        merge_commands = [
+            "merge",
+            "merge this",
+            "merge this pr",
+        ]
+        cancel_commands = [
+            "cancel",
+            "cancel merge",
+            "cancel the merge",
+            "stop",
+            "stop merge",
+            "stop the merge",
+        ]
+
+        def parse_action(s: str) -> Optional[str]:
+            if any(f"@tvm-bot {c}" in s for c in merge_commands):
+                return "merge"
+
+            if any(f"@tvm-bot {c}" in s for c in cancel_commands):
+                return "cancel"
+
+            return None
+
+        # Check regular comments
+        all_comments = []
+        for comment in self.raw["comments"]["nodes"]:
+            all_comments.append((comment["updatedAt"], comment["body"]))
+
+        # Check top-level review comments
+        for review in self.reviews():
+            all_comments.append((review["updatedAt"], review["body"]))
+
+        all_comments = sorted(all_comments, key=lambda x: x[0])
+        actions = [parse_action(body) for _, body in all_comments]
```

Comment:
just following up here--did we want to add a filter for who can merge? i think that is a good idea.

Reply:
added along with a test